tests/virtualization/xen/ssh_guests_init.pm

```diff
@@ -23,22 +23,30 @@ use testapi;
 use utils;
 
 sub run {
-    my ($self) = @_;
     select_console 'root-console';
     opensusebasetest::select_serial_terminal();
-    my $hypervisor = get_required_var('QAM_XEN_HYPERVISOR');
-    my $domain     = get_required_var('QAM_XEN_DOMAIN');
+    my $hypervisor = get_required_var('HYPERVISOR');
 
     foreach my $guest (keys %xen::guests) {
         record_info "$guest", "Establishing SSH connection to $guest";
-        assert_script_run "ssh root\@$hypervisor 'ping -c3 -W1 $guest.$domain'";
-        assert_script_run "ssh-keyscan $guest.$domain >> ~/.ssh/known_hosts";
-        exec_and_insert_password "ssh-copy-id root\@$guest.$domain";
+
+        # Fill the current pairs of hostname & address into /etc/hosts file
+        assert_script_run "ssh root\@$hypervisor 'sed -i \"/$guest/d\" /etc/hosts'";
```

Comment:
Current test structure should be using ipmi(hypervisor)+qemu(client). If possible, would you please share your plan for supporting the client tests on  ipmi(hypervisor) directly?

Reply:
Yes, in my opinion this can be running whole on IPMI. My plan is to make wrapper for the `ssh root\@$hypervisor` so it will be skipped on IPMI but otherwise I see no other issues. Do you have some other suggestions?

Reply:
I support your way :+1: